src/ReportPortal.Shared/Internal/Delegating/ExponentialRetryRequestExecuter.cs

```diff
@@ -97,7 +101,9 @@ public override async Task<T> ExecuteAsync<T>(Func<Task<T>> func, Action<Excepti
                     else
                     {
                         TraceLogger.Error($"Error while invoking '{func.Method.Name}' method. Current attempt: {i}.\n{exp}");
-                        throw;
+                        exceptions.Add(new HttpRequestException("Something unexpected happened. Limit of retries has been reached.", exp));
```

Comment:
`Something unexpected happened` - be more specific here as well.

Reply:
fixed.